tests/acceptance/guid-file/registration-file-detail-test.ts

```diff
@@ -294,4 +295,109 @@ module('Acceptance | guid file | registration files', hooks => {
+            'File metadata resource language is properly updated.');
+    });
+
+    test('View-only section does not update', async function(this: ThisTestContext, assert) {
+        await visit(`/--file/${this.file.id}`);
+
+        await click('[data-test-edit-metadata-button]');
+        // Update title
+        await fillIn('[data-test-title-field] > div > textarea', 'A New Title');
+        // Update description
+        await fillIn('[data-test-description-field] > div > textarea', 'A New Description');
+        // Update resource type
+        await selectChoose('[data-test-select-resource-type]', 'InteractiveResource');
+        // Update resource language
+        await selectChoose('[data-test-select-resource-language]', 'Latin');
+        // Cancel changes
+        await click('[data-test-cancel-metadata-button]');
+
+        assert.dom('[data-test-file-title]').doesNotHaveTextContaining('A New Title',
+            'Cancel metadata edit button properly working for title.');
+        assert.dom('[data-test-file-description]').doesNotHaveTextContaining('A New Description',
+            'Cancel metadata edit button properly working for description.');
+        assert.dom('[data-test-file-resource-type]').doesNotHaveTextContaining('InteractiveResource',
+            'Cancel metadata edit button properly working for resource type.');
+        assert.dom('[data-test-file-language]').doesNotHaveTextContaining('Latin',
+            'Cancel metadata edit button properly working for resource language.');
+    });
+
+    test('Error and canceling edit', async function(this: ThisTestContext, assert) {
+        setupOnerror((e: any) => assert.ok(e, 'Error is handled'));
+
+        const node = server.create('node', {
+            id: 'mtadt',
+            currentUserPermissions: [Permission.Read, Permission.Write],
+        });
+        const projectFile = server.create('file', {
+            target: node,
+            name: 'Test File',
+        });
+        const url = `/--file/${projectFile.id}`;
+
+        server.namespace = '/v2';
+        server.patch('/custom_file_metadata_records/:id', () => ({
+            errors: [{
+                detail: 'Could not patch metadata',
+                source: { pointer: 'points to nowhere' },
+            }],
+        }), 400);
+
+        await visit(url);
+
+        await click('[data-test-edit-metadata-button]');
+        await fillIn('[data-test-title-field] > div > textarea', 'A New Title');
+        await click('[data-test-save-metadata-button]');
+        assert.dom('#toast-container', document as any)
+            .hasTextContaining(t('osf-components.file-metadata-manager.error-saving-metadata'));
+        await click('[data-test-cancel-metadata-button]');
+
+        server.get('/custom_file_metadata_records/:id', _ => {
+            assert.dom('[data-test-file-title]').doesNotHaveTextContaining('A New Title',
+                'Canceling edit after server error working properly.');
+        });
```

Comment:
This isn't the proper check to see if the title remained the same. `server.get('/custom_file_metadata_records/:id', ....` is reassigning the mirage server's behavior to run this function when this endpoint is hit. This isn't necessary, since we don't want to overwrite the GET custom_file_metadata_record. The proper check would be to `assert.dom('[selector-for-the-title]').hasText('The original title value');`

Reply:
There might be some misunderstanding here as to what `server.get()` does. When you do `server.get('url_path', handler)`, you are telling mirage server to use the `handler` function to handle the `GET` request sent to `url_path`. If there is already a handler defined in `config.ts` for that endpoint, it will be overridden. The same thing with `server.patch()` and `server.post()`.

Therefore, when you do
```
server.get('/custom_file_metadata_records/:id', _ => {
            assert.dom('[data-test-file-title]').doesNotHaveTextContaining('A New Title',
                'Canceling edit after server error working properly.');
        });
```
you are saying that the mirage server should return a test assertion when a request is sent to `/custom_file_metadata_records/:id`. The test might run and pass, but it is not really overriding the API functionality correctly, and therefore not testing the app properly.

Reply:
Thank you Yuhuai, we spoke about this earlier.